src/main/java/org/bonitasoft/connectors/rest/RESTConnector.java

```diff
@@ -412,8 +412,15 @@ private void setOutputs(final HttpResponse response, Request request) throws IOE
                         if (contentType != null
                                 && !Strings.isNullOrEmpty(contentType.getValue())
                                 && contentType.getValue().toLowerCase().contains("json")) {
-                            setBody(new ObjectMapper().readValue(bodyResponse,
-                                    bodyResponse.startsWith("[") ? List.class : HashMap.class));
+                            if (bodyResponse.startsWith("[")) {
```

Comment:
I would use a try/catch instead of an if statement here.
Something like
```
try{
      setBody(new ObjectMapper().readValue(bodyResponse,
                            bodyResponse.startsWith("[") ? List.class : HashMap.class));
}catch (JsonParseException | JsonMappingException e) {
     LOGGER.warning(String.format(
                     "bodyAsObject output cannot be set. Response content is not valid json(%s).",
                      bodyResponse));
 }
```

This way it covers more failures cases.

Reply:
Made a new commit with try/catch AND the if statment because I also want to handle the case of simple values.

Reply:
👍